stdlib/public/Concurrency/AsyncThrowingStream.swift

```diff
@@ -12,61 +12,203 @@
 
 import Swift
 
+/// An asynchronous sequence generated from an error-throwing closure that
+/// calls a continuation to produce new elements.
+///
+/// `AsyncThrowingStream` conforms to `AsyncSequence`, providing a convenient
+/// way to create an asynchronous sequence without manually implementing an
+/// asynchronous iterator. In particular, an asynchronous stream is well-suited
+/// to adapt callback- or delegation-based APIs to participate with
+/// `async`/`await`.
+///
+/// In contrast to `AsyncStream`, this type can throw an error from the awaited
+/// `next()`, which terminates the stream with the thrown error.
+///
+/// You initialize an `AsyncThrowingStream` with a closure that receives an
+/// `AsyncThrowingStream.Continuation`. Produce elements in this closure, then
+/// provide them to the stream by calling the continuation's `yield(_:)` method.
+/// When there are no further elements to produce, call the continuation's
+/// `finish()` method. This causes the sequence iterator to produce a `nil`,
+/// which terminates the sequence. If an error occurs, call the continuation's
+/// `finish(throwing:)` method, which causes the iterator's `next()` to throw
+/// the error to the awaiting call point. The continuation is `Sendable`, which
+/// permits calling it from concurrent contexts external to
+/// the iteration of the `AsyncThrowingStream`.
+///
+/// An arbitrary source of elements can produce elements faster than they are
+/// consumed by a caller iterating over them. Because of this, `AsyncThrowingStream`
+/// defines a buffering behavior, allowing the stream to buffer a specific
+/// number of oldest or newest elements. By default, the buffer limit is
+/// unbounded, that is, `Int.max`.
+///
+/// ### Adapting Existing Code to Use Streams
+///
+/// To adapt existing callback code to use `async` / `await`, use the callbacks
+/// to provide values to the stream, by using the continuation's `yield(_:)`
+/// method.
+///
+/// Consider a hypothetical `QuakeMonitor` type that provides callers with
+/// `Quake` instances every time it detects an earthquake. To receive callbacks,
+/// callers set a custom closure as the value of the monitor's
+/// `quakeHandler` property, which the monitor calls back as necessary. Callers
+/// can also set an `errorHandler` to receive asychronous error notifications,
+/// such as the monitor service suddenly becoming unavailable.
+///
+///     class QuakeMonitor {
+///         var quakeHandler: ((Quake) -> Void)?
+///         var errorHandler: ((Error) -> Void)?
+///
+///         func startMonitoring() {…}
+///         func stopMonitoring() {…}
+///     }
+///
+/// To adapt this to use `async`-`await`, extend the `QuakeMonitor` to add a
+/// `quakes` property, of type `AsyncThrowingStream<Quake>`. In the getter for
+/// this property, return an `AsyncThrowingStream`, whose `build` closure --
+/// called at runtime to create the stream -- does the following:
+///
+/// 1. Creates a `QuakeMonitor` instance.
```

Comment:
This should probably be a bulleted list instead of a numbered list.

Reply:
Similar to line 62 of `AsyncStream.swift`, I think we prefer numbered lists when there's a sequence of steps, performed in order. In any case, we should do both of these the same way, whatever we decide.